### PR TITLE
Identity facade + Okta lab

### PR DESCRIPTION
**Pull request by Teodelas**
Any value in having these steps in the DevRel repo?

I'm a Googler: teodlh@google.com

### CODE REVIEW

File: labs/idp-okta-integration/readme.md
Comment (danistrebel) [inline labs/idp-okta-integration/readme.md:1]: Please rename this file to README.md to be consistent with the rest of devrel

File: labs/idp-okta-integration/readme.md
Comment (danistrebel) [inline labs/idp-okta-integration/readme.md:10]: please change to the upstream identity-facade

File: labs/idp-okta-integration/readme.md
Comment (danistrebel) [inline labs/idp-okta-integration/readme.md:56]: Officially called `Apigee UI` now :)

File: labs/idp-okta-integration/readme.md
Comment (danistrebel) [inline labs/idp-okta-integration/readme.md:107]: Please make this part dynamic and explain how to get the values

**Review by danistrebel — CHANGES_REQUESTED** [review]
Great PR Teo! I added a few preliminary comments and asked @JoelGauci to also take a look as he owns the identity-facade.

In the meantime can you please take a look at the CONTRIBUTING file. We'll need you to add an entry to the root REAME.md file to explain the lab, add yourself to the CODEOWNERs file and create a pipeline.sh file. 

Additionally it would be great if you could follow the [claat generation](https://github.com/apigee/devrel/blob/main/labs/bdd-proxy-development/generate-docs.sh) as used by the other labs to make your your lab is also nicely rendered like here: https://apigee.github.io/devrel/labs/bdd-proxy-development/

I also triggered the linter so you can see its output in the PR checks.

Thanks again and please do reach out to any of us if you have any questions.

File: labs/idp-okta-integration/readme.md
Comment (JoelGauci) [inline labs/idp-okta-integration/readme.md:155]: Okta does not redirect to https://httpbin.org/get
Okta redirects to the Apigee callback URL (/v1/oauth20/callback), which controls the incoming query parameters, generate an authorization code (using the same value as the one provided by Okta) and performs a redirection on the client app redirect_uri (here https://httpbin.org/get) providing the authorization_code and initial state parameters

File: labs/idp-okta-integration/readme.md
Comment (JoelGauci) [inline labs/idp-okta-integration/readme.md:165]: this is an example for OAUTH_CODE value. It would be worth mentioning it

File: labs/idp-okta-integration/readme.md
Comment (danistrebel) [reply labs/idp-okta-integration/readme.md:1]: Up to you if you want to follow the existing labs and have a lab.md and a README.md file or but both in the same.

**Comment by Teodelas** [comment]
I've made the edits and added an additional step to show how to secure the Hello-World API using the Identity Facade.

**Comment by danistrebel** [comment]
Thanks @Teodelas, please also add the changes in #419 to this PR and add the lab to the devrel main readme file here: https://github.com/apigee/devrel/blob/main/README.md?plain=1#L77-L81

**Comment by danistrebel** [comment]
pipeline failed with error 

```
Step #3 - "Run Pipeline": [INFO] DevRel Pipeline: labs/idp-okta-integration
Step #3 - "Run Pipeline": labs/idp-okta-integration (err): /usr/bin/run-pipelines.sh: line 31: ./pipeline.sh: Permission denied
Step #3 - "Run Pipeline": [INFO] DevRel Pipeline done: labs/idp-okta-integration
```

Please check if the pipeline.sh file is executable: `chmod +x pipeline.sh`

**Comment by danistrebel** [comment]
@Teodelas very close to green now. 

1. Please see Comment https://github.com/apigee/devrel/pull/417#issuecomment-962887440 about the root README entry
1. Please take a look at the markdown lint errors reported in https://github.com/apigee/devrel/runs/4161994697?check_suite_focus=true

**Comment by Teodelas** [comment]
> @Teodelas very close to green now.
> 
> 1. Please see Comment [Identity facade + Okta lab #417 (comment)](https://github.com/apigee/devrel/pull/417#issuecomment-962887440) about the root README entry
> 2. Please take a look at the markdown lint errors reported in https://github.com/apigee/devrel/runs/4161994697?check_suite_focus=true

I think I've addressed these.

File: labs/idp-okta-integration/README.md
Comment (danistrebel) [inline labs/idp-okta-integration/README.md:161]: add sh

e.g.

```sh
foo
```

File: labs/idp-okta-integration/README.md
Comment (danistrebel) [inline labs/idp-okta-integration/README.md:238]: we'll have to fence this code block (to make sure the httpbin.org links aren't validated)


```sh
...
```

File: labs/idp-okta-integration/README.md
Comment (danistrebel) [inline labs/idp-okta-integration/README.md:254]: we'll have to fence this code block (to make sure the httpbin.org links aren't validated)


```sh
...
```

**Review by danistrebel — CHANGES_REQUESTED** [review]
Hi Teo,

1. We're still missing the repo root README.md entry. Please have a look at the earlier comments.

2. The linter had issues validating two httpbin.org links that are part of form params. I added the fix to this review and kindly ask you to disable the checks for the two code blocks.

3. Optional (some won't work for claat like the very first comment) but encouraged to take a look at the markdown lint comments below. Especially the code block language indicator helps with displaying the syntax highlighting of the code.
```
README.md:1 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "id: idp-okta-integration"]
README.md:9 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:10:81 MD013/line-length Line length [Expected: 80; Actual: 232]
README.md:12 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:26 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:29:81 MD013/line-length Line length [Expected: 80; Actual: 90]
README.md:31:81 MD013/line-length Line length [Expected: 80; Actual: 127]
README.md:34 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Prerequisites"]
README.md:35 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- [Signup](https://developer.o..."]
README.md:36:81 MD013/line-length Line length [Expected: 80; Actual: 111]
README.md:49 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:52 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Retrieve Apigee URL"]
README.md:53:81 MD013/line-length Line length [Expected: 80; Actual: 124]
README.md:56 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "1. Open the [Apigee Portal](ht..."]
README.md:59:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
README.md:59:81 MD013/line-length Line length [Expected: 80; Actual: 104]
README.md:63 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:64 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Create Okta User"]
README.md:65 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "1. Sign into your Okta develop..."]
README.md:68:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
README.md:73 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Create Okta App"]
README.md:74 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "1. In the left hand side navig..."]
README.md:76:81 MD013/line-length Line length [Expected: 80; Actual: 115]
README.md:78 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "4. In the next screen, configu..."]
README.md:79 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- App integration name: Apigee..."]
README.md:82:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 5]
README.md:84:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 5]
README.md:85:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 5]
README.md:88 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "![Assignments](assets/okta-ass..."]
README.md:89:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 5; Style: 1/1/1]
README.md:89 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "5. Click **Save** to be taken ..."]
README.md:93 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:93 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:99 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:104 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:108:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 2; Style: 1/1/1]
README.md:110 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:112 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:113:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 3; Style: 1/1/1]
README.md:113 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "3. Run the following commands ..."]
README.md:115 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:120 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:121:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 4; Style: 1/1/1]
README.md:121 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "4. Verify the variables have b..."]
README.md:123 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:132:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 5; Style: 1/1/1]
README.md:136:81 MD013/line-length Line length [Expected: 80; Actual: 166]
README.md:137 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "![idp facade output](assets/id..."]
README.md:138:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 6; Style: 1/1/1]
README.md:138 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "6. Copy and save the following..."]
README.md:144:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 7; Style: 1/1/1]
README.md:146 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Base64 encoding of client id..."]
README.md:147 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:147 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:153 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:156:81 MD013/line-length Line length [Expected: 80; Actual: 162]
README.md:158:81 MD013/line-length Line length [Expected: 80; Actual: 88]
README.md:158 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "1. Open an incognito tab in a ..."]
README.md:159:81 MD013/line-length Line length [Expected: 80; Actual: 99]
README.md:159 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- This step simulates an appli..."]
README.md:161 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:161 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "     ```"]
README.md:162:81 MD013/line-length Line length [Expected: 80; Actual: 197]
README.md:166:81 MD013/line-length Line length [Expected: 80; Actual: 130]
README.md:166:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 2; Style: 1/1/1]
README.md:170:81 MD013/line-length Line length [Expected: 80; Actual: 374]
README.md:170:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 3; Style: 1/1/1]
README.md:170 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "3. After successful authentica..."]
README.md:170:289 MD034/no-bare-urls Bare URL used [Context: "https://httpbin.org/get"]
README.md:171:81 MD013/line-length Line length [Expected: 80; Actual: 152]
README.md:171 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- In a real-world scenario, th..."]
README.md:175:81 MD013/line-length Line length [Expected: 80; Actual: 154]
README.md:175:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 4; Style: 1/1/1]
README.md:175 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "4. Pass the authorization code..."]
README.md:176 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:176 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:178:81 MD013/line-length Line length [Expected: 80; Actual: 134]
README.md:187 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:188:81 MD013/line-length Line length [Expected: 80; Actual: 99]
README.md:188:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 5; Style: 1/2/3]
README.md:188 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "5. Apigee will respond with a ..."]
README.md:190 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:191:81 MD013/line-length Line length [Expected: 80; Actual: 124]
README.md:191:1 MD029/ol-prefix Ordered list item prefix [Expected: 2; Actual: 6; Style: 1/2/3]
README.md:193 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:196 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:201:81 MD013/line-length Line length [Expected: 80; Actual: 247]
README.md:212:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 5]
README.md:213:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 5]
README.md:214:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 5]
README.md:220:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 3; Style: 1/1/1]
README.md:220 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "3. Under Public, click on Apps"]
README.md:221:81 MD013/line-length Line length [Expected: 80; Actual: 116]
README.md:221 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Notice there is an app creat..."]
README.md:222:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 4; Style: 1/1/1]
README.md:222 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "4. Click on +App and configure..."]
README.md:223 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Name: Hello World App"]
README.md:226:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 5]
README.md:227:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 5]
README.md:229 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:232:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 5; Style: 1/2/3]
README.md:233:1 MD029/ol-prefix Ordered list item prefix [Expected: 2; Actual: 6; Style: 1/2/3]
README.md:237:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 7; Style: 1/1/1]
README.md:237 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "7. Update the environment vari..."]
README.md:238 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:238 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:242:81 MD013/line-length Line length [Expected: 80; Actual: 169]
README.md:244 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:247:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 8; Style: 1/1/1]
README.md:247 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "8. Like before, visit the $AUT..."]
README.md:248 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Authenticate as testuser@tes..."]
README.md:253:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 9; Style: 1/1/1]
README.md:253 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "9. Run through the remaning au..."]
README.md:254 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:254 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:256:81 MD013/line-length Line length [Expected: 80; Actual: 183]
README.md:270 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:271 MD022/blanks-around-headings/blanks-around-headers Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Summary"]
README.md:272:81 MD013/line-length Line length [Expected: 80; Actual: 146]
README.md:274 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
```

**Comment by Teodelas** [comment]
Fixed all Lint errors except this one:
https://github.com/DavidAnson/markdownlint/blob/v0.24.0/doc/Rules.md#md041
- Says "id: idp-okta-integration" at the top of the readme.md is a violation.

**Comment by danistrebel** [comment]
Thats' cool. We're ignoring MD041 anyways https://github.com/apigee/devrel/blob/main/.markdown-lint.yml#L19 :)
 
 
 I ran the static checks again here: https://github.com/apigee/devrel/runs/4688708777?check_suite_focus=true
 
 The entry in the root README.md (see comment above) is still missing therefore the `
DevRel Static Checks Pipeline / Repo Structure ` check is still failing. Can you add an item to the list that links to your lab and give it a one sentence summary?

**Comment by danistrebel** [comment]
Very close. Two things

1. there is a typo in your codeowners path ipd-okta-integration -> **idp**-okta-integration
2. our linter checks the links in the markdown. This is handy but unfortunately raises an issue with the labs. Can you please remove the github.io link in the main readme until your solution is merged and available on github.io? We'll create a link in the readme for you once the PR is in.

Much appreciated!

**Review by danistrebel — APPROVED** [review]

File: labs/idp-okta-integration/README.md
Comment (JoelGauci) [inline labs/idp-okta-integration/README.md:252]: Product

Comment (danistrebel) [reply labs/idp-okta-integration/README.md:252]: @Teodelas please see the trivial typo above. After that we're good to merge. Thanks in advance!

**Comment by danistrebel** [comment]
Thanks a bunch @Teodelas. Great lab! 🎉

**Pull request merged**
